### DISCUSSION TITLE
How to 'move' data from an old redash server to a new server

### DISCUSSION THREAD

**Discussion by qq20230530** [getredash/redash#6871] [Q&A] [answered]
Hi,
Could anyone tell me how to 'move' data from an old redash server to a new server?
We were using redash built on an EC2 server, used amazon AMI to create the server.
And now we have a new server, located in our office.
How to 'get' all the data on the old server, and 'move' them on the new server?

**Comment by justinclift** [comment, marked as answer]
It's fairly straight forward, so you shouldn't have too much trouble. :smile:

Your Redash installation has three pieces, all generally stored under `/opt/redash` on your Redash server:

1. The `docker-compose.yml` file
   * Contains the details of what version of the main components to use, where they're stored, and so on.
   * Stuff like "what version of PostgreSQL is being used", "what version of Redash is being used", etc.

2. A docker "environment" file
   * Generally named `env`
   * It's a plain text file, structured as `key=value` pairs, one per line
   * Has things like the database password and encryption keys in it

3. The internal Redash database
   * This is a PostgreSQL database
   * Generally stored in a directory structure named `postgres-data`
   * Safe to copy around *if* the database is shut down when you're doing the copy
   * But if you copy this while the database is still running, the copy isn't going to be any good

Probably the simplest approach is to transfer your backup of the above pieces to the new server, store them at the same location there, then start Redash from the command line on the new server.

Do you have ssh access to your new server, and are you familiar with using ssh? :smile:

**Reply by qq20230530** [reply]
Thank you so much!!
I WILL have ssh access to the new server. 
Now we are having another company build the server for us.

I am concerned about one thing.
Since the old server is on amazon. But our new server is different.
I am not sure which they are using, docker? or something else? but surely not amazon AMI, cause we found out it was too expensive to use EC2 server.
Actually it was me who built the old server, following the instructions on the page below.
https://redash.io/help/open-source/setup#aws
So, I wonder whether all the related files are at the same locations?

And, one more question, maybe too basic, forgive me. I am not an IT expert, just started working as an engineer.
On our existing redash page, we wrote many complicated queries,  want to use them on the new server, too, if possible.
Will those queries also remain there after the transfer?
I mean, after transferring all the files you mentioned, when I open the redash page running on the new server, I will be able to see all the queries, right?
Or, is anything else needed?

**Reply by justinclift** [reply]
Heh Heh Heh

I have lots of questions now. :smile:

Which company is building the new server for you, and how much are they charging?  Is it an ongoing hosting fee thing?

Do they specialise in Redash, or just server building, or something else?

Also, do you know the specifications of the server they're setting up for you?  It'll need about 8GB (minimum) to run reliably well, and more than that if you've got heaps of stuff already loaded.

> On our existing redash page, we wrote many complicated queries, want to use them on the new server, too, if possible.

Yeah, that's why I was saying above that you need to restore a backup of those three pieces to the new server.  If you do that, then all of your saved queries will be there.

---

Oh, if you're uncomfortable sharing some of those details publicly, you're welcome to email me directly (`justin.clift@newdash.io`) as I'm really interested in Redash hosting competitors. :smile:

---

Regardless of what you do, DO NOT kill your old EC2 server until you've had the new one running for while and are happy with it... just in case.  Btw, how much was your old EC2 server costing you per month?

**Reply by qq20230530** [reply]
Thank you for your advice about data transfer.
And thank you for teaching me useful information.
Regarding the new server thing, I will have a meeting with my colleague next Wednesday, and will definitely get back to you on email!!!

**Reply by justinclift** [reply]
All good.  If you're open to alternative hosting for Redash, my company ([Newdash.io](https://newdash.io)) does that. :smile:

Still haven't gotten around to putting up a proper website, it's still just a placeholder for the moment.

Pricing wise, we're using [the same pricing levels](http://web.archive.org/web/20200306092851/https://redash.io/pricing/) as the original Redash.io online service.  Those figures are US$.
